### PR TITLE
nixos/gdm: Fix pulseaudio tmpfiles structure

### PR DESCRIPTION
**Pull request by tobim**
Fix the following startup failure of the sound service in the gdm
session that was introduced by #75893:
```
Feb 16 11:44:15 qp pulseaudio[1432]: W: [pulseaudio] core-util.c: Failed to open configuration file '/run/gdm/.config/pulse//daemon.conf': Not a directory
Feb 16 11:44:15 qp pulseaudio[1432]: W: [pulseaudio] daemon-conf.c: Failed to open configuration file: Not a directory
Feb 16 11:44:15 qp systemd[1380]: pulseaudio.service: Main process exited, code=exited, status=1/FAILURE
Feb 16 11:44:15 qp systemd[1380]: pulseaudio.service: Failed with result 'exit-code'.
Feb 16 11:44:15 qp systemd[1380]: Failed to start Sound Service.
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/x11/display-managers/gdm.nix [hunk 1]
```diff
@@ -168,7 +168,8 @@ in
     systemd.tmpfiles.rules = [
       "d /run/gdm/.config 0711 gdm gdm -"
     ] ++ optionals config.hardware.pulseaudio.enable [
-      "L+ /run/gdm/.config/pulse - - - - ${pulseConfig}"
+      "d /run/gdm/.config/pulse 0711 gdm gdm"
```
Comment (worldofpeace) [inline nixos/modules/services/x11/display-managers/gdm.nix]: this is different than `"d /run/gdm/.config 0711 gdm gdm -"`

Comment (tobim) [reply nixos/modules/services/x11/display-managers/gdm.nix]: Omitting the age filed has the same effect as `-` according to the [man page](https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html).
Or are you referring to something else?

File: nixos/modules/services/x11/display-managers/gdm.nix [hunk 2]
```diff
@@ -168,7 +168,8 @@ in
     systemd.tmpfiles.rules = [
       "d /run/gdm/.config 0711 gdm gdm -"
```
Comment (worldofpeace) [inline nixos/modules/services/x11/display-managers/gdm.nix]:
```suggestion
      "d /run/gdm/.config 0711 gdm gdm"
```

File: nixos/modules/services/x11/display-managers/gdm.nix (hunk 1 above)
Comment (worldofpeace) [reply nixos/modules/services/x11/display-managers/gdm.nix]: sorry, https://github.com/NixOS/nixpkgs/pull/80274/files#r379931503 for consistency

**Comment by worldofpeace** [comment]
Made a clean vm and looked for the file
```
[root@nixos:/run/gdm/.config/pulse]# ll
total 20
-rw------- 1 gdm  gdm  108 Feb 23 12:22 8169178e55f3457b8553dd899e413214-card-database.x86_64-pc-linux-gnu.simple
-rw------- 1 gdm  gdm    1 Feb 23 12:22 8169178e55f3457b8553dd899e413214-default-sink
-rw------- 1 gdm  gdm    1 Feb 23 12:22 8169178e55f3457b8553dd899e413214-default-source
-rw------- 1 gdm  gdm  459 Feb 23 12:22 8169178e55f3457b8553dd899e413214-device-volumes.x86_64-pc-linux-gnu.simple
-rw------- 1 gdm  gdm  256 Feb 23 12:22 cookie
lrwxrwxrwx 1 root root  54 Feb 23 12:22 default.pa -> /nix/store/df6p9q81pjrv73ijq7yzvprmp2xpiggi-default.pa
```

**Review by worldofpeace — APPROVED** [review]

**Comment by worldofpeace** [comment]
backported to 20.03 in 7a15ea936d9e395a98e98866199edb4f4ccb939f

**Comment by tobim** [comment]
Thanks!

**Comment by mmahut** [comment]
I'm getting the following on today's switch to unstable, probably related?

```
Failed to open directory 'pulse': Not a directory
symlink(/nix/store/df6p9q81pjrv73ijq7yzvprmp2xpiggi-default.pa, /run/gdm/.config/pulse/default.pa) failed: Not a directory
warning: error(s) occurred while switching to the new configuration```

**Comment by tobim** [comment]
Did you get that message from the gdm session after boot or during `nixos-rebuild`?

**Comment by mmahut** [comment]
During ```nixos-rebuild```, the switch did not happen as it failed.

**Comment by tobim** [comment]
This is probably because of the `switch` causing the `tmpfiles` rules to be re-applied. On your running system the path `/run/gdm/.config/pulse` is a symlink pointing to the generated `default.pa`, so the command
```
 "d /run/gdm/.config/pulse 0711 gdm gdm" 
```
fails.

The workaround is to use `nixos-rebuild boot` this time around instead.

**Comment by tobim** [comment]
Or, if you can't reboot: Just delete the symlink manually.

**Comment by worldofpeace** [comment]
Yeah, /run isn't going to stick. This would only be an issue for unstable users. 19.09 -> 20.03 should be fine.

**Pull request merged**